Update HcalSimpleDigiAndRecProducer.cxx

tillrue opened the pull request:
In the energy weighted position average, the cumulative energy was used instead of the individual hit energy. This resulted in the odd distribution of RecHits that you can see in issue #1681 . The energy weighted time average is affected as well.

Further, I have implemented the algorithm for the side Hcal bars but I am happy to receive feedback on how to make the code more efficient or on which parts could be left out which is why I open this PR as a draft.

### What are the issues that this addresses?
This resolves issue #1681 . There you can see how the current code is producing RecHits that are much further spread out than the SimHits which might be due to using the cumulative deposited energy instead of the individual one for each hit to do the weighting. On the issue page, you can also see the same event with the implemented code.

## Check List
- [✅] I successfully compiled _ldmx-sw_ with my developments.
- [✅] I read, understood and follow the [coding rules](https://ldmx-software.github.io/developing/coding-rules.html).
<!-- If these coding rules are confusing or you need help, still open the PR as a _draft_ and we can help you! -->
- [✅] I ran my developments and the following shows that they are successful.
See issue #1681 for the result of the fix.

tillrue committed:
Update HcalSimpleDigiAndRecProducer.cxx

In the energy weighted position average, the cumulated energy was used instead of the individual hit energy. This resulted in the odd distribution of RecHits. The time weighted average is affected as well.

Further, I have implemented the algorithm for the side Hcal bars but I am happy to receive feedback on how to make the code more efficient or on which parts could be left out.

github-actions[bot] committed:
Apply clang-format

EinarElen commented on Hcal/src/Hcal/HcalSimpleDigiAndRecProducer.cxx:
```diff
@@ -119,10 +120,30 @@ void HcalSimpleDigiAndRecProducer::produce(framework::Event& event) {
+      }
+      // This might be irrelevant
+      else {
+        xpos = stripCenter.x();
+        ypos = stripCenter.y();
+        zpos = stripCenter.z();
```
Probably worth adding a warning here so that if someone makes some new hcal model and things change, they'd be alerted

tillrue replied:
Good idea and probably better than exception. I'm not sure how to create a warning within ldmx-sw though but maybe one of the reviewers can help with that?

bryngemark replied:
I think an exception is ok until someone makes a change that allows us to end up here somehow, and then that should be dealt with accordingly. 

An example of printing stuff to the logger (includes terminal messages; use this instead of `cout`) can be found here: https://github.com/LDMX-Software/ldmx-sw/blob/3a1f52b2207aa75cc81ffc62c6a82402f716c58c/Hcal/src/Hcal/HcalVetoProcessor.cxx#L138-L139 
in your case select a level of more severity, like warning or fatal

tillrue replied:
I see. Will implement this. Thanks for the help!

EinarElen reviewed (approved):
Looks correct :)

tillrue committed:
Add incode comments HcalSimpleDigiAndRecProducer.cxx

Here, I added incode comments for better readability. 

Further, it might be a good idea to raise an exception if none of the known bar orientations is found. However, I'm not sure if this is necessary or even how to "artificially" trigger this exception as all hits usually have an associated bar orientation.

github-actions[bot] committed:
Apply clang-format

tillrue committed:
Update HcalSimpleDigiAndRecProducer.cxx

Now a warning message is included in case no proper bar orientation could be found.

github-actions[bot] committed:
Apply clang-format
